src/v/cluster/types.h

```diff
@@ -1582,6 +1598,11 @@ struct topic_properties
     std::optional<std::chrono::milliseconds> flush_ms;
     std::optional<size_t> flush_bytes;
 
+    // introduced to drive validation at recovery time, when .recovery=true and
+    // .remote_topic_properties.has_value() the main use case is to have a
+    // manually editable knob in the topic_manifest json file
+    std::optional<recovery_checks> recovery_checks{std::nullopt};
```

Comment:
@andijcr this is a good question. can you motivate the decision?

Reply:
removed as discussed.

some excerpts from the discussion, as reference: 

"
Originally the idea was to be able to control checks by editing the topic_manifest.json . But with topic_manifest moving to serde that's not really an option anymore, so i can see how there is no reason for them to be in topic_properties
"

"
They were in topic_properties because the design was to give the option to use a different validation strategy per topic, instead of having only one per cluster. i see topic_manifest as a serialized topic_configuration, so it had the added bonus (once everything was in place) of being able to set offline the validation strategy for a topic by editing it.
The original design had the idea of leaving part of the validation to the node recreating the final partition, but that part was removed in the end
"

"
[validation] is done in the shard creating the create_topic command, but one possible extension was to just perform a manifest existence check and leave a deeper scrub to the shard managing the partition
"